stl/inc/atomic

```diff
@@ -472,6 +506,27 @@ struct _Atomic_storage<_Ty, 1> { // lock-free using 1-byte intrinsics
         return false;
     }
 
+    void wait(_Ty _Expected, const memory_order _Order = memory_order_seq_cst) const noexcept {
+        long _Spin_context = 0;
+        for (;;) {
+            const char _Expected_bytes = _Atomic_reinterpret_as<char>(_Expected);
+            const char _Observed_bytes = load(_Order);
+            if (_Expected_bytes != _Observed_bytes) {
+                break;
+            }
+            _Atomic_wait_direct(_STD addressof(_Storage), _STD addressof(_Expected), 1, _Spin_context);
+        }
+        _Atomic_unwait_direct(_STD addressof(_Storage), _Spin_context);

```

Comment:
Should probably introduce a macro (or publicize `_STL_WIN32_WINNT` in `yvals.h`) to remove these `_Atomic_unwait_direct` calls when we know we don't need the fallback (e.g. ARM64 is always Win10, so no fallback extra function call needed)

Reply:
One way to approach this issue is to move everything into .cpp, including variable re-check.

One thing that stopped me is the fact that with such move that loads are type-erased  it is harder to handle cases when _value representation_ is different from _object representation_, such cases as:

```
struct eleven_bits {
   std::uint16_t value : 11;
   // and 5 bits undefined
};

using atomic_uint11_t = std::atomic<eleven_bits>;
```

In fact, I think such case is already broken for `compare_exchange_strong`  / `compare_exchange_weak`. 

On the other hand, an implementation that handles this case should deal with the fact that whatever `WaitOnAddress` takes is type-erased, and still should use `WaitOnAddress` directly for such type. So apparently this should be handled by detecting this situation, and creating new comparand out of unused bits of source and used bits of passed comparand.

So, will still consider moving into .cpp most of the stuff.

Reply:
I see there is already #23 for it, and it is blocked by compiler support.
So will have to implement this for now using _object representation_.

Reply:
Decided to avoid extra call by exposing the flag for it.

Reply:
Actually, done to the best way it could be. On x86 and x64 it is up to the user to define _WIN32_WINNT